validphys2/src/validphys/closuretest/multiclosure_output.py

```diff
@@ -88,11 +201,93 @@ def datasets_bias_variance_ratio(datasets_expected_bias_variance, each_dataset):
     """
     records = []
     for ds, (bias, var, ndata) in zip(each_dataset, datasets_expected_bias_variance):
-        records.append(dict(dataset=str(ds), ndata=ndata, ratio=bias / var))
+        records.append(
+            dict(
+                dataset=str(ds),
+                ndata=ndata,
+                ratio=bias / var,
+                sqrt_ratio=np.sqrt(bias / var),
+            )
+        )
     df = pd.DataFrame.from_records(
-        records, index="dataset", columns=("dataset", "ndata", "ratio")
+        records, index="dataset", columns=("dataset", "ndata", "ratio", "sqrt_ratio")
     )
-    df.columns = ["ndata", "bias/variance"]
+    df.columns = ["ndata", "bias/variance", "sqrt(bias/variance)"]
+    return df
+
+
+@table
+def datasets_bias_variance(datasets_expected_bias_variance, each_dataset):
+    """For each dataset calculate the expected bias and expected variance
+    across fitsband tabulate the results. Bias and Variance are normalized by number of data points
+
+    Notes
+    -----
+
+    This is to check the weight each dataset/process has in the calculation of the complete R_bv ratio.
+    This is because one dataset alone could have a correct B/V=1 but if Bias and Variance are both centered
+    around a number >> 1 this means that in the calculation of B/V total ratio the specific dataset/
+    process is going to have much more weight than the rest
+
+    """
+    records = []
+    for ds, (bias, var, ndata) in zip(each_dataset, datasets_expected_bias_variance):
+        records.append(dict(dataset=str(ds), ndata=ndata, bias=bias, variance=var))
+    df = pd.DataFrame.from_records(
+        records, index="dataset", columns=("dataset", "ndata", "bias", "variance")
+    )
+    df.columns = ["ndata", "bias", "variance"]
+    return df
+
+
+@figure
+def plot_bias_variance_ndata(datasets_expected_bias_variance, each_dataset):
+    """
+    plot the trend of bias and variance with the number of points. It is expected that all these lie
+    on a line
+    """
+    data = []
+    biases = []
+    variances = []
+    for bias, var, ndata in zip(datasets_expected_bias_variance):
+        data.append(ndata)
+        biases.append(bias)
+        variances.append(var)
+    fig, ax = plotutils.subplots()
+    ax.plot(ndata, biases, label="biases")
+    ax.plot(ndata, variances, label="variances")
+    ax.legend()
+
+    return fig
+
+
+@table
+def bias_variance_ratio_by_process(datasets_expected_bias_variance, each_dataset):
+    """
+    Same as `datasets_bias_variance_ratio` but adding another index to table, namely,
+    the name of the process
+    """
+
+    process_type = []
+
+    for ds, (bias, var, ndata) in zip(each_dataset, datasets_expected_bias_variance):
+        # open plot file and get process name
+        plt_file = ds.commondata.plotfiles[1]
+        with open(plt_file, "r") as file:
+            card = yaml.safe_load(file)
+        prcs = card["nnpdf31_process"]
```

Comment:
There are easier, correct, ways to get the process. See what is done in config.py.